mne/preprocessing/_beer_lambert_law.py

```diff
@@ -0,0 +1,119 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+import re as re
+import numpy as np
+
+import mne
+from .. import rename_channels
+from ..io import BaseRaw
+from ..io.pick import _picks_to_idx
+from ..io.constants import FIFF
+from ..utils import _validate_type
+
+
+def beer_lambert_law(raw, ppf=0.1):
+    r"""Convert NIRS optical density data to haemoglobin concentration.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The optical density data.
+    ppf : float
+        The partial pathlength factor.
+
+    Returns
+    -------
+    raw : instance of Raw
+        The modified raw instance.
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'raw')
+
+    freqs = np.unique(_channel_frequencies(raw))
+    picks = _check_channels_ordered(raw, freqs)
+    abs_coef = _load_absorption(freqs)
+    distances = raw._probe_distances()
```

Comment:
Based on how the code currently works, I think this will only succeed if the raw is an instance of `RawNIRX`. This is an oversight in our design. This `_probe_distances` should probably just be made into a method of `BaseRaw`, or maybe better (since it's really kind of data specific) a private function that takes a `raw` instance as its first argument).

The reason for this is that we want this to work even if someone has done:
```
mne.io.read_raw_nirx(fname).save('temp_raw.fif')
raw = mne.io.read_raw_fif('temp_raw.fif')
raw_h = beer_lambert_law(raw)
```

Reply:
I think I have addressed this in 7baa40f 
But I am unsure if `_beer_lambert_law.py` is the best place to put this, as its probably something that will be commonly used. Should I create a file with all useful fNIRS specific functions like this? If so, where would it be best placed? Or is it ok in here?